### PR TITLE
feat: add winning token on transfer

### PR DESCRIPTION
**Pull request by Daniel-16**
@tarzann419 This is my final push for the day. 
To explain this; when a user transfers a token that is a winning token to another user, it adds the token to the new wallet's account
I also added a scheduler for checking users with three winning tokens

### CODE REVIEW

**Comment by tarzann419** [comment]
Truly truly, amazing stuff Dano @Daniel-16 
Just to be sure, after the user transfers the token, do you pop it out of their collected tokens?

Also this is the summary; subject to corrections:
	1.	Weekly Token Assignment: Every week, a scheduler adds a special “winningToken” to an item from the available tokens
	2.	User Purchase: When a user buys an item with a winningToken, that token is added to their profile under “collectedTokens”
	3.	Admin Notification: Another scheduler checks if a user has collected three winningTokens and alerts the admin with the user’s ID and username
	4.	Token Transfer: If a user transfers a winningToken to another user, the token is added to the new user’s profile
	5.	Token Collection Check: A scheduler also verifies users with three winningTokens and performs necessary actions

tl;dr: The system in a nutshell, rewards users for collecting winning tokens, tracks these tokens, and keeps the admin informed(for now)

**Comment by Daniel-16** [comment]
So yes, everything is the summary correctly.
Also I haven't added the feature to pop yet.
But that collected tokens field to make our work easier, only contains tokens that are the winning tokens. ONLY
I'll add the function for popping winning token from a user if such user transfers a winning token from their collectedTokens
For now, I go to sleep against today
